### PR TITLE
[CCFPCM-378] Report First Page

### PR DESCRIPTION
**Pull request by chelsea-EYDS**
[CCFPCM-0378](https://bcdevex.atlassian.net/browse/CCFPCM-0378)

**Objective:** 

Generate first page of the report
Add conditional styling

Screenshot 2023-03-14 at 2 54 28 PM

### CODE REVIEW

File: apps/backend/src/excelexport/excelexport.service.ts [hunk 1]
```diff
@@ -53,20 +54,60 @@ export class ExcelexportService {
...
+    this.workbook.title = title;
+    this.workbook.created = new Date(format(new Date(), 'yyyy, MM, dd'));
+  }
+
   public addSheet(name: string): void {
     this.workbook.addWorksheet(name);
   }
 
+  /* eslint-disable  */
+  public addHeader(sheetName: string, style: any): void {
```
Comment (vyasworks) [inline apps/backend/src/excelexport/excelexport.service.ts:67]: `addHeaderToRow`

File: apps/backend/src/excelexport/excelexport.service.ts [hunk 2]
```diff
@@ -53,20 +54,60 @@ export class ExcelexportService {
...
+    const sheet = this.workbook.getWorksheet(sheetName);
+    sheet.insertRow(1, []);
+    const row = sheet.getRow(1);
+    row.addPageBreak();
+    row.getCell('E').value = sheetName;
+    row.getCell('E').style = style;
+    row.height = 20;
+    sheet.getRow(2).addPageBreak();
```
Comment (vyasworks) [inline apps/backend/src/excelexport/excelexport.service.ts:75]: make dynamic based on input!

File: apps/backend/src/excelexport/excelexport.service.ts [hunk 3]
```diff
@@ -53,20 +54,60 @@ export class ExcelexportService {
...
     this.workbook.addWorksheet(name);
   }
 
+  /* eslint-disable  */
+  public addHeader(sheetName: string, style: any): void {
+    const sheet = this.workbook.getWorksheet(sheetName);
+    sheet.insertRow(1, []);
+    const row = sheet.getRow(1);
+    row.addPageBreak();
+    row.getCell('E').value = sheetName;
```
Comment (vyasworks) [inline apps/backend/src/excelexport/excelexport.service.ts:72]: determine cell / cells based on input

File: apps/backend/src/excelexport/excelexport.service.ts [hunk 4]
```diff
@@ -53,20 +54,60 @@ export class ExcelexportService {
...
+    row.getCell('E').style = style;
+    row.height = 20;
+    sheet.getRow(2).addPageBreak();
+  }
+
+  /*eslint-disable @typescript-eslint/no-unused-vars*/
+  public addCellStyle(
```
Comment (vyasworks) [inline apps/backend/src/excelexport/excelexport.service.ts:79]: is it adding style to a cell or a row???

File: apps/backend/src/excelexport/excelexport.service.ts [hunk 5]
```diff
@@ -53,20 +54,60 @@ export class ExcelexportService {
     }
   }
 
+  public generateWorkbook(title: string): void {
```
[the remark this answers is not in the document]
Comment (chelsea-EYDS) [reply apps/backend/src/excelexport/excelexport.service.ts:57]: resolved in report_two PR

File: apps/backend/src/excelexport/excelexport.service.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (chelsea-EYDS) [reply apps/backend/src/excelexport/excelexport.service.ts:72]: resolved in report_two

File: apps/backend/src/excelexport/excelexport.service.ts (hunk 1 above)
Comment (chelsea-EYDS) [reply apps/backend/src/excelexport/excelexport.service.ts:67]: The header is added to the sheet here ie:
Screenshot 2023-03-16 at 11 10 29 AM

File: apps/backend/src/excelexport/excelexport.service.ts (hunk 2 above)
Comment (chelsea-EYDS) [reply apps/backend/src/excelexport/excelexport.service.ts:75]: resolved in report_two

File: apps/backend/src/excelexport/excelexport.service.ts (hunk 3 above)
Comment (chelsea-EYDS) [reply apps/backend/src/excelexport/excelexport.service.ts:72]: resolved in report_two

File: apps/backend/src/excelexport/excelexport.service.ts (hunk 4 above)
Comment (chelsea-EYDS) [reply apps/backend/src/excelexport/excelexport.service.ts:79]: resolved in report_two renamed to "addRowStyle"

File: apps/backend/src/excelexport/excelexport.service.ts (hunk 3 above)
Comment (chelsea-EYDS) [reply apps/backend/src/excelexport/excelexport.service.ts:72]: Passing in a variable from the reporting service to set the header placement

File: apps/backend/src/excelexport/excelexport.service.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (chelsea-EYDS) [reply apps/backend/src/excelexport/excelexport.service.ts:57]: renamed to "addWorkbookMetadata"

File: apps/backend/src/excelexport/excelexport.service.ts (hunk 1 above)
Comment (chelsea-EYDS) [reply apps/backend/src/excelexport/excelexport.service.ts:67]: resolved?

**Review by vyasworks — APPROVED** [review]

**Pull request merged**